src/components/utils/events.ts

```diff
@@ -1,31 +1,34 @@
 import { isEmpty } from '../utils';
 
 /**
- * @class EventDispatcher
+ * Mapped type with subscriptions list
  *
- * Has two important methods:
+ * event name -> array of callbacks
+ */
+type Subscriptions<EventMap> = {
```

Comment:
```suggestion
type Subscriptions<EventMap extends Record<string, any>> = {
```

Reply:
This seems to be still relevant

Reply:
why do we need it?

Reply:
it will lead returning extension of `Record<string, any>` in all event maps

Reply:
To specify EventMap is an object